src/openms/source/ANALYSIS/OPENSWATH/DIAPrescoring.cpp

```diff
@@ -127,55 +127,63 @@ namespace OpenMS
+    int chg = 1;
+    for (const auto& transition : lt)
+    {
+      if (transition.fragment_charge != 0) chg = transition.fragment_charge;
+      DIAHelpers::addIsotopes2Spec(transition.getProductMZ(), transition.getLibraryIntensity(), spectrumWIso, nr_isotopes_, chg);
+    }
+    spectrumWIsoNegPreIso.resize(spectrumWIso.size());
+    std::copy(spectrumWIso.begin(), spectrumWIso.end(), spectrumWIsoNegPreIso.begin());
+    double totalNegWeight = 0.5; // how much of ONE transition should be negatively weighted at the prePeaks (distributed equally on them)
+    UInt nrNegPeaks = 2;
+    for (const auto& transition : lt)
+    {
+      if (transition.fragment_charge != 0) chg = transition.fragment_charge;
+      DIAHelpers::addPreisotopeWeights(transition.getProductMZ(), spectrumWIso, nrNegPeaks, 0.0);
+      DIAHelpers::addPreisotopeWeights(transition.getProductMZ(),
+                                       spectrumWIsoNegPreIso,
+                                       nrNegPeaks,
+                                       -totalNegWeight/(double(lt.size() * nrNegPeaks)));
+    }
+    DIAHelpers::sortByFirst(spectrumWIso);
+    DIAHelpers::sortByFirst(spectrumWIsoNegPreIso);
+    // compare against the spectrum with 0 weight preIsotope peaks
+    std::vector<double> mzTheor, intTheor;
+    DIAHelpers::extractFirst(spectrumWIso, mzTheor);
+    DIAHelpers::extractSecond(spectrumWIso, intTheor);
     std::vector<double> intExp, mzExp;
-    DIAHelpers::integrateWindows(spec, theomasses, dia_extract_window_, intExp, mzExp);
+    DIAHelpers::integrateWindows(spec, mzTheor, dia_extract_window_, intExp, mzExp);
     std::transform(intExp.begin(), intExp.end(), intExp.begin(), OpenSwath::mySqrt());
-    std::transform(theorint.begin(), theorint.end(), theorint.begin(), OpenSwath::mySqrt());
-
-    double intExptotal = std::accumulate(intExp.begin(), intExp.end(), 0.0);
-    double intTheorTotal = std::accumulate(theorint.begin(), theorint.end(), 0.0);
-
-    OpenSwath::normalize(intExp, intExptotal, intExp);
-    OpenSwath::normalize(theorint, intTheorTotal, theorint);
+    std::transform(intTheor.begin(), intTheor.end(), intTheor.begin(), OpenSwath::mySqrt());
 
-    manhattan = OpenSwath::manhattanDist(intExp.begin(), intExp.end(), theorint.begin());
+    double intExpTotal = std::accumulate(intExp.begin(), intExp.end(), 0.0);
+    double intTheorTotal = std::accumulate(intTheor.begin(), intTheor.end(), 0.0);
 
-    //std::cout << spectrum.size() << std::endl;
-    DIAHelpers::addPreisotopeWeights(firstIstotope, spectrum2, 2, -0.5);
-    std::vector<double>  theorint2;
-    DIAHelpers::extractSecond(spectrum, theorint2);
-    std::transform(theorint2.begin(), theorint2.end(), theorint2.begin(), OpenSwath::mySqrt());
+    OpenSwath::normalize(intExp, intExpTotal, intExp);
+    OpenSwath::normalize(intTheor, intTheorTotal, intTheor);
 
-    intExptotal = OpenSwath::norm(intExp.begin(), intExp.end());
```

Comment:
This used euclidean norm on something that was previously normalized through sum?!